pkg/che/che.go

```diff
@@ -23,7 +40,135 @@ type Client struct {
+	reqData := url.Values{}
+	reqData.Set("name", username)
+	res, err := c.cheRequest(http.MethodGet, cheUserFindPath, reqData)
+	if err != nil {
+		return false, errors.Wrapf(err, "request to find Che user '%s' failed", username)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode == http.StatusOK {
+		return true, nil
+	} else if res.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+	resBody, readError := rest.ReadBody(res.Body)
+	if readError != nil {
+		log.Error(readError, "error while reading body of the find Che user response")
+	}
+	return false, errors.Errorf("request to find Che user '%s' failed, Response status: '%s' Body: '%s'", username, res.Status, resBody)
+}
+
+// GetUserIDByUsername returns the user ID that maps to the given username or an error if the user was not found or there was a problem with the request
+func (c *Client) GetUserIDByUsername(username string) (string, error) {
+	reqData := url.Values{}
+	reqData.Set("name", username)
+	res, err := c.cheRequest(http.MethodGet, cheUserFindPath, reqData)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get Che user ID for user '%s'", username)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode != http.StatusOK {
+		resBody, readError := rest.ReadBody(res.Body)
+		if readError != nil {
+			log.Error(readError, "error while reading body of the get Che user ID response")
+		}
+		err = errors.Errorf("unable to get Che user ID for user '%s', Response status: '%s' Body: '%s'", username, res.Status, resBody)
+		return "", err
+	}
+	cheUser, err := readCheUser(res)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get Che user ID for user '%s'", username)
+	}
+	return cheUser.ID, err
+}
+
+// DeleteUser deletes the Che user with the given user ID
+func (c *Client) DeleteUser(userID string) error {
+	log.Info("Deleting user", "userID", userID)
+	res, err := c.cheRequest(http.MethodDelete, path.Join(cheUserPath, userID), nil)
+	if err != nil {
+		return errors.Wrapf(err, "unable to delete Che user with ID '%s'", userID)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode != http.StatusNoContent && res.StatusCode != http.StatusNotFound {
+		resBody, readError := rest.ReadBody(res.Body)
+		if readError != nil {
+			log.Error(readError, "error while reading body of the delete Che user response")
+		}
+		err = errors.Errorf("unable to delete Che user with ID '%s', Response status: '%s' Body: '%s'", userID, res.Status, resBody)
+	} else if res.StatusCode == http.StatusNotFound {
+		log.Info("The user was not deleted because they were not found", "userID", userID)
+	}
+	return err
+}
+
+func (c *Client) cheRequest(rmethod, endpoint string, queryParams url.Values) (*http.Response, error) {
+	// get Che route URL
+	cheURL, err := route.GetRouteURL(c.k8sClient, c.config.GetCheNamespace(), c.config.GetCheRouteName())
+	if err != nil {
+		return nil, err
+	}
+
+	// create request
+	req, err := http.NewRequest(rmethod, cheURL+endpoint, nil)
```

Comment:
shouldn't the param be just `method`? It's just a detail - you can ignore it as well ;-) 
```suggestion
func (c *Client) cheRequest(method, endpoint string, queryParams url.Values) (*http.Response, error) {
	// get Che route URL
	cheURL, err := route.GetRouteURL(c.k8sClient, c.config.GetCheNamespace(), c.config.GetCheRouteName())
	if err != nil {
		return nil, err
	}

	// create request
	req, err := http.NewRequest(method, cheURL+endpoint, nil)
```

Reply:
It can be, I have no problem changing it 🙂